bndtools.core/src/bndtools/central/Central.java

```diff
@@ -89,85 +92,60 @@
 	private static final Supplier<EclipseWorkspaceRepository>	eclipseWorkspaceRepository	= Memoize
 		.supplier(EclipseWorkspaceRepository::new);
 
-	private static Auxiliary									auxiliary;
-
 	static final AtomicBoolean									indexValid					= new AtomicBoolean(false);
 
-	private final BundleContext									bundleContext;
 	private final Map<IJavaProject, Project>					javaProjectToModel			= new HashMap<>();
 	private final List<ModelListener>							listeners					= new CopyOnWriteArrayList<>();
 
 	private RepositoryListenerPluginTracker						repoListenerTracker;
-	private final InternalPluginTracker							internalPlugins;
+	private InternalPluginTracker							internalPlugins;
 
 	@SuppressWarnings("unused")
 	private static WorkspaceRepositoryChangeDetector			workspaceRepositoryChangeDetector;
 
 	private static RepositoriesViewRefresher					repositoriesViewRefresher	= new RepositoriesViewRefresher();
-	private static BundleContext								context;
 	private static ServiceRegistration<Workspace>				workspaceService;
+	private static BundleContext								context;
 
-	static {
-		try {
-			context = FrameworkUtil.getBundle(Central.class)
-				.getBundleContext();
-			Bundle bndlib = FrameworkUtil.getBundle(Workspace.class);
-			auxiliary = new Auxiliary(context, bndlib);
-
-		} catch (Exception e) {
-			// ignore
-		}
-	}
-
-	/**
-	 * WARNING: Do not instantiate this class. It must be public to allow
-	 * instantiation by the Eclipse registry, but it is not intended for direct
-	 * creation by clients. Instead call Central.getInstance().
-	 */
-	@Deprecated
-	public Central() {
-		bundleContext = FrameworkUtil.getBundle(Central.class)
-			.getBundleContext();
-		internalPlugins = new InternalPluginTracker(bundleContext);
-
-	}
-
-	@Override
-	public void start() {
+	@Activate
+	void activate(BundleContext bc) {
+		context = bc;
 		instance = this;
-
-		repoListenerTracker = new RepositoryListenerPluginTracker(bundleContext);
+		repoListenerTracker = new RepositoryListenerPluginTracker(bc);
 		repoListenerTracker.open();
+		internalPlugins = new InternalPluginTracker(bc);
 		internalPlugins.open();
+		// Trigger building of the Workspace object and registering it a service
+
+		promiseFactory().submit(Central::getWorkspace);
```

Comment:
It's in a promise because my understanding was that `@Activate` methods should run quickly and that `getWorkspace()` might take time to initialize. If this is not a requirement I'm happy to inline the call. Otherwise I'll add the `onFailure()`.

Reply:
A Promise is generally used to obtain the result of a calculation in the future. Here you do not even keep the returned promise :-) But, as I said, you can use the returned promise to log any error that occurs. You should do that.

Reply:
Done.